### ISSUE TITLE
"Explore" menu listing games as "Unknown", despite games having an associated database entry with values for the filter criteria

### ISSUE THREAD

**Issue by jackcmedia** [libretro/RetroArch#12576]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description
"Explore" menu lists games as having "unknown" developer, release year, genre and franchise, despite having associated a database entry with most of those details available.

On RetroArch 1.9.5, I have the following games scanned into my collection, with the following details being available:
- Final Fantasy VI Advance (USA) (Nintendo - Gameboy Advance)
  - Developer: TOSE
  - Release Year: 2007
  - Genre: RPG
  - Franchise: Final Fantasy
- Parasite Eve 1 (USA) (Sony - Playstation, both disks)
  - Developer: Squaresoft
  - Release Year: 1998
  - Genre: Survival Horror
  - Franchise: N/A
- Vagrant Story (USA) (Sony - Playstation)
  - Developer: Squaresoft
  - Release Year: 2000
  - Genre: Action
  - Franchise: N/A

Given the above details, I'd expect that the PSX games are correctly listed in the "Explore" menu under their associated "Developer", "Release Year" and "Genre". The only time I expect the games to be listed as "Unknown" is when listing them by "Franchise".

However, all PSX games are listed under "Unknown" except when filtering by "System Name".

### Expected behavior
When using the explore function to check the scanned games in my collection, I get the following results:
- By Developer: "TOSE" (FFVI) and "Squaresoft" (both PSX games)
- By Release Year: "2007" (FFVI) , "1998" (Parasite Eve 1, both disks) and "2000" (Vagrant Story)
- By Genre: "RPG" (FFVI), "Survival Horror" (Parasite Eve 1, both disks) and "Action" (Vagrant Story)
- By Franchise: "Final Fantasy" (FFVI) and "Unknown" (both PSX games)

### Actual behavior
When using the explore function to check the scanned games in my collection, I get the following results:
- By Developer: "TOSE" (FFVI) and **"Unknown" (both PSX games)**
- By Release Year: "2007" (FFVI) and **"Unknown" (both PSX games)**
- By Genre: "RPG" (FFVI) and **"Unknown" (both PSX games)**
- By Franchise: "Final Fantasy" (FFVI) and "Unknown" (both PSX games)

### Steps to reproduce the bug
1. Scan rom directory containing the games.
2. Verify that all games scanned have an associated database entry with Developer, Release Year, Genre and/or Franchise.
3. Use the "Explore" menu to filter by "Developer", "Release Year", "Genre" and/or "Franchise".
4. Games filtered by "Developer", "Release Year", "Genre" or "Franchise" show as "Unknown" despite having those values in their database entries.

### Bisect Results
N/A, I don't know how to do this

### Version/Commit
- RetroArch: 1.9.5 x64 (Git a129764)

### Environment information
- OS: Windows 10 Pro x64

### Screenshots
- Final Fantasy VI Advance (USA) database entry:
![image](https://user-images.githubusercontent.com/15652751/123510382-65aba000-d67b-11eb-983e-ec16fa279e7f.png)
- Parasite Eve (USA) (Disc 1) database entry:
![image](https://user-images.githubusercontent.com/15652751/123510241-a2c36280-d67a-11eb-9f4e-f0aab8530840.png)
- Parasite Eve (USA) (Disc 2) database entry:
![image](https://user-images.githubusercontent.com/15652751/123510274-c686a880-d67a-11eb-9aba-4cfe090d9f8f.png)
- Vagrant Story (USA) database entry:
![image](https://user-images.githubusercontent.com/15652751/123510290-d9997880-d67a-11eb-9732-c2e2f13afd52.png)

- All games:
![image](https://user-images.githubusercontent.com/15652751/123510308-f5048380-d67a-11eb-97d6-a3c11091da69.png)

- Filter by Developer (not expected):
![image](https://user-images.githubusercontent.com/15652751/123510332-22513180-d67b-11eb-86a2-faf4e26c35c0.png)
- Filter by Release Year (not expected):
![image](https://user-images.githubusercontent.com/15652751/123510336-2a10d600-d67b-11eb-9d44-9961d977e375.png)
- Filter by Genre (not expected):
![image](https://user-images.githubusercontent.com/15652751/123510345-3137e400-d67b-11eb-9e41-6eb88a061eeb.png)
- Filter by Franchise (expected):
![image](https://user-images.githubusercontent.com/15652751/123510354-46ad0e00-d67b-11eb-9a71-bb366a575eb1.png)
- Filter by System Name (expected):
![image](https://user-images.githubusercontent.com/15652751/123510361-4f054900-d67b-11eb-82aa-6026beca9da5.png)

**Comment by jackcmedia** [comment]
Interesting, I tried adding more PSX games (Saiyuki - Journey West, being one of them) and those are correctly populating the various filters of the "Explore" menu.

**Comment by seraphlive** [comment]
Similar issue occurred to me when I scanned a folder with a almost full set of nes/famicom roms. In the "Explore" menu, the release date only shows 1986~1990, while I definitely have roms released since 1983. Most roms are under the Unknown section.

I also checked one rom in the Unknown section for its information and it can properly be linked to the database entry.

I'll try to completely clear the play list file and try to import all the roms again later to see if the problems still exist.

**Comment by seraphlive** [comment]
I tried again and found that Famicom disk system roms I imported work fine. Only metadata of NES ROMs can be displayed in "Explore" tab.

Screenshot of 1.9.6 macOS metal version:
<img width="931" alt="image" src="https://user-images.githubusercontent.com/12388970/125291288-cb727a00-e353-11eb-9c2a-b8a5d9011345.png">

Also found this issue #11275 back from 2020. Looks like they are related.

**Comment by jackcmedia** [comment]
While debugging the code at [1] found that, for both `Vagrant Story (USA)` and `Parasite Eve (USA)`, no metadata is returned from `Sony - Playstation.rdb`, a database which holds the metadata of PSX games.

Inspecting said database, found that both `Vagrant Story (USA)` and `Parasite Eve (USA)` are listed twice:
- `Vagrant Story (USA)` has serial numbers `SLUS-01040` and `SLUS-01040GH` (the latter is a Greatest Hits rerelease).
- `Parasite Eve (USA)` has serial numbers `SLUS-00662` and `SLUS-00662-0` (no idea what the latter is supposed to be).

`SLUS-01040GH` and `SLUS-00662-0` are listed first on the database, and neither have the metadata that I was expecting to find (developer, publisher, franchise, year of release, etc). The other ones, `SLUS-01040` and `SLUS-00662`, are listed towards the bottom and *do* contain the metadata.

I'm thinking that I probably have the rereleases and, when fetching the details, processing stops at the first match. This results in the "Unknown" values due to missing metadata, which is curious since reading the database entry from within RetroArch shows the complete metadata and both serial numbers (failed to notice this when first reporting the issue).

Not sure if the same occurs for @seraphlive with the NES roms.

[1] https://github.com/libretro/RetroArch/blob/master/menu/menu_explore.c#L402-L741

**Comment by jackcmedia** [comment]
Some progress:

I downloaded and compiled my own RDBs, by following the instructions at [1]. Along the way, I found that the `Sony - Playstation.dat` from REDUMP [2] is the only source that lists both `Vagrant Story (USA)` and `Parasite Eve (USA)` twice.

After removing the duplicate entries from said source, compiling `Sony - Playstation.rdb` and running RetroArch with it, the result is that both games can be correctly filtered using the "Explore" function.

Sample RDB with `Vagrant Story (USA)` (`SLUS-01040GH`) removed:
[Sony - PlayStation.zip](https://github.com/libretro/RetroArch/files/7187036/Sony.-.PlayStation.zip)

Filter by Developer - `Vagrant Story (USA)` is now listed under `Squaresoft`:
![image](https://user-images.githubusercontent.com/15652751/133821982-2df240c8-1c44-4f57-82ee-70735de2cddf.png)

Filter by Publisher - `Vagrant Story (USA)` is now listed under `Squaresoft`:
![image](https://user-images.githubusercontent.com/15652751/133822035-ce02bc57-6114-4e62-bb8a-067e4c55499d.png)

Filter by Release Year - `Vagrant Story (USA)` is now listed under `2000`:
![image](https://user-images.githubusercontent.com/15652751/133822086-c108f5bd-9ce5-47e4-8132-aa87206a0cf8.png)

Filter by Genre - `Vagrant Story (USA)` is now listed under `Action`:
![image](https://user-images.githubusercontent.com/15652751/133822117-32e46770-9a6e-4735-b07d-2a5cd28096ae.png)


I lack the expertise to implement a solution, but I can think of a few proposals:
- Have the "Explore" function continue parsing database entries to pull the missing metadata. This is already done by the "Database Entry" function.
- Modify the database so that it links from the metadata-less entries to the ones with metadata, provided that they have the same CRC-32.


[1] https://github.com/libretro/RetroArch/tree/master/libretro-db#compiling-all-rdbs-with-libretro-build-databasesh
[2] https://github.com/libretro/libretro-database/blob/b706aa6877ca2f82db7b4e6a5216da0ac13f85ad/metadat/redump/Sony%20-%20PlayStation.dat

---

TL;DR:
- "Explore" function lists some PlayStation games under "Unknown" value for some filters, despite Retroarch showing the missing metadata when looking at the game's database entry.
- Sample affected games are `Vagrant Story (USA)` (`SLUS-01040`) and `Parasite Eve (USA)` (`SLUS-00662`).
- Finding 1: `Sony - Playstation.rdb` lists said games twice, with REDUMP being the source of the duplicates.
- Finding 2: "Explore" only fetches the metadata of the first matching entry.
- Finding 3: Removing the duplicates from the REDUMP source and compiling my own RDBs, the issue doesn't manifest.

**Comment by zoltanvb** [comment]
Fixed at least partially by #18778
